editor/src/clj/editor/code/view.clj

```diff
@@ -2428,6 +2460,48 @@
         (ui/event-handler e
           (handle-input-method-changed! view-node e))))))
 
+(defn- hover-view [canvas {:keys [hovered-element layout lines]}]
+  (let [r ^Rect (first (data/cursor-range-rects layout lines (:region hovered-element)))
+        anchor (.localToScreen canvas (.-x r) (.-y r))]
+    {:fx/type fxui/with-popup
+     :desc {:fx/type fxui/ext-value :value canvas}
+     :showing true
+     :anchor-x (.getX anchor)
+     :anchor-y (.getY anchor)
+     :anchor-location :window-bottom-left
+     :auto-hide true
+     :auto-fix true
+     :hide-on-escape true
+     :consume-auto-hiding-events true
+     :content [{:fx/type :label
+                :stylesheets [(str (io/resource "dialogs.css"))]
+                :style-class ["label" "hover-popup"]
+                :max-width 300
+                :wrap-text true
+                :text (str (:message (:region hovered-element)))}]}))
+
+(defn- create-hover! [view-node canvas]
+  (let [state (atom nil)
+        refresh (fn refresh []
+                  (g/with-auto-evaluation-context evaluation-context
+                    (let [hovered-element (g/node-value view-node :hovered-element evaluation-context)]
+                      (if (and (= :region (:type hovered-element))
+                               (= :diagnostic (:type (:region hovered-element))))
```

Comment:
Perhaps instead of hard-coding it to `:diagnostic` regions, we could associate a `:hover-view` function similar to `:on-click!` that returns a cljfx map for us to display here?

Reply:
Hmm yes, I like the idea of decomplecting hovers from diagnostics. Will do!